node/stateHandler.go

```diff
@@ -0,0 +1,159 @@
+package node
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/rand"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/nknorg/nkn/v2/chain"
+	"github.com/nknorg/nkn/v2/chain/store"
+	"github.com/nknorg/nkn/v2/common"
+	"github.com/nknorg/nkn/v2/pb"
+	"github.com/nknorg/nkn/v2/util/log"
+)
+
+// NewGetStateMessage creates a GET_STATES message
+func NewGetStateMessage(hashes []common.Uint256) (*pb.UnsignedMessage, error) {
+	msgStates := make([]*pb.StateRequest, len(hashes))
+	for i, _ := range hashes {
+		msgStates[i] = &pb.StateRequest{Hash: hashes[i].ToArray()}
+	}
+	states := &pb.GetStates{
+		Id:   rand.Uint64(),
+		Reqs: msgStates,
+	}
+
+	buf, err := proto.Marshal(states)
+	if err != nil {
+		return nil, err
+	}
+
+	msg := &pb.UnsignedMessage{
+		MessageType: pb.MessageType_GET_STATES,
+		Message:     buf,
+	}
+
+	return msg, nil
+}
+
+// NewGetStatesReply creates a GET_STATES_REPLY message in respond to GET_STATES
+// message
+func NewGetStatesReply(nodes [][]byte) (*pb.UnsignedMessage, error) {
+	msgNodes := make([]*pb.StateNode, len(nodes))
+	for i, n := range nodes {
+		msgNodes[i] = &pb.StateNode{Node: n}
+	}
+
+	msgBody := &pb.GetStatesReply{
+		Nodes: msgNodes,
+	}
+
+	buf, err := proto.Marshal(msgBody)
+	if err != nil {
+		return nil, err
+	}
+
+	msg := &pb.UnsignedMessage{
+		MessageType: pb.MessageType_GET_STATES_REPLY,
+		Message:     buf,
+	}
+
+	return msg, nil
+}
+
+func (localNode *LocalNode) getStatesMessageHandler(remoteMessage *RemoteMessage) ([]byte, bool, error) {
+	replyMsg, err := NewGetStateMessage(nil)
+	if err != nil {
+		return nil, false, err
+	}
+	replyBuf, err := localNode.SerializeMessage(replyMsg, false)
+	if err != nil {
+		return nil, false, err
+	}
+	msgBody := &pb.GetStates{}
+	err = proto.Unmarshal(remoteMessage.Message, msgBody)
+	if err != nil {
+		return nil, false, err
```

Comment:
The first parameter of all return since this line should be `replyBuf` instead of `nil`